src/dashboard/Data/Browser/Browser.react.js

```diff
@@ -320,9 +321,21 @@ export default class Browser extends DashboardView {
     return data;
   }
 
+  async fetchParseDataCount(source, filters) {
+    const query = queryFromFilters(source, filters);
+    const count = await query.count({ useMasterKey: true });
```

Comment:
I think - let's merge it for now and look at optimizations in a separate PR, unless you want to try anything different?